### ISSUE TITLE
Possible Read Temperature Implementation Error Causes Lack of Precision

### ISSUE THREAD

**Issue by nickcounts** [xinabox/SW01#3]
Hi, All,

I was investigating some unexpected temperature data quantization. We expected to see variation on the order of 0.01C (per the datasheet), but we were actually seeing a steps of around 0.16C. This got me curious, so I dug into the library a little and I think there may be a copy/paste error (and a << vs >>) mixup.

![image](https://user-images.githubusercontent.com/10506020/36747294-a9450882-1bc2-11e8-9813-a0a604ff5b18.png)

## The Currently Released Codebase

```c++
void xSW01::readTemperature(void)
{
    int32_t var1, var2;
    
    int32_t rawTemp = ((uint32_t)xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_MSB) << 12);
	rawTemp |= ((uint32_t)xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_MSB) << 4);
	rawTemp |= ((xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_MSB) << 4) & 0x0F);
  
```

## Proposed Change:

```c++
  void xSW01::readTemperature(void)
{
    int32_t var1, var2;
    
    int32_t rawTemp =   ((uint32_t)xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_MSB) << 12);
            rawTemp |=  ((uint32_t)xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_CSB) << 4);
            rawTemp |= (((uint32_t)xCore.read8(BME280_I2C_ADDRESS, BME280_REG_TEMP_LSB) >> 4) & 0x0F);


```

## Rationale:

It appears that we were reading the Most Significant Bit register 3 times, and attempting to shift it to it's correct position for a 12 bit representation. The first two `rawTemp` lines did that (except for reading the same register). The third `rawTemp` line doesn't actually modify `rawTemp`

After doing a `<<` shift of 4, it is bitwise `&` with `0x0F`, which is `00001111` so it throws away whatever bits were just shifted and keeps the empty (zeros) portion.

Interestingly, with this behavior, the smallest change we could expect (neglecting a change in the calibration values) would be `00010000` which is 16 in decimal! That corresponds to the 0.16C quantization we were observing.

Hope this helps!

Cheers

**Comment by brendanvanbreda** [comment]
@nickcounts  great find, the above has been implemented in 89f331076e1fc1e8906f47e7d9533d4e20fb5818

**Comment by nickcounts** [comment]
@brendanvanbreda 

Thanks :)  I just wanted to double check something in your commit [89f3310](https://github.com/xinabox/xSW01/commit/89f331076e1fc1e8906f47e7d9533d4e20fb5818)

I _think_ line 150 may need to be a >> instead of a <<    - I'm not in a place to test, so I could be totally wrong! Just wanted to double-check 👍
